lib/array_intersection.rb

```diff
@@ -1,3 +1,3 @@
 def intersection(list1, list2)
-  
+  list2.select &list1.to_set.method(:include?)
```

Comment:
Nice work using method reference. Depending on the context, I would often prefer a more verbose solution that makes each step of the algorithm a bit more explicit. Were an interview candidate to produce this code I'd ask them to step through the order of execution carefully to ensure they understood it well and were not creating unnecessary loops or objects.